src/mono/mono/mini/mini-codegen.c

```diff
@@ -2286,10 +2286,21 @@ mono_opcode_to_cond (int opcode)
 	case OP_CMOV_LGT_UN:
 		return CMP_GT_UN;
 	default:
+		return (CompRelation)-1;
+	}
+}
+
+CompRelation
+mono_opcode_to_cond (int opcode)
+{
+	CompRelation rel = mono_try_opcode_to_cond (opcode);
+
+	if (rel == (CompRelation)-1) {
 		printf ("%s\n", mono_inst_name (opcode));
 		g_assert_not_reached ();
 		return (CompRelation)0;
```

Comment:
This lane will never be reached.

Reply:
Some compilers can't detect that and they will error out.

Reply:
nit: Now that we're C11 we can probably redefine `G_GNUC_NORETURN` as

```c
#if __STDC_VERSION__ >= 202311L
#define G_GNUC_NORETURN [[noreturn]]
#else
#define G_GNUC_NORETURN _Noreturn
#endif
```

and then all compilers will know that `mono_assertion_message_unreachable` doesn't return

Reply:
https://github.com/dotnet/runtime/pull/93131